### PR TITLE
Overhaul terminal commands

### PR DESCRIPTION
**Pull request by checkraisefold**
## About the Pull Request

adds all the previously un-included terminal commands in the folder in, also fixed all of them
(this does not change maps. dreammaker made a backup folder in maps folder that I didn't notice and had to delete after.)

## Why It's Good For The Game

terminal hacking with ntnet just became 10x more fun

## Changelog

:cl:
add: A whole lot of modular computer terminal commands.
/:cl:

### CODE REVIEW

**Comment by TerryTehCoder** [comment]
I swear these were all in a bit ago before. Did they get removed in a past PR intentionally? Unintentionally? They disappeared very randomly.

**Comment by checkraisefold** [comment]
> I swear these were all in a bit ago before. Did they get removed in a past PR intentionally? Unintentionally? They disappeared very randomly.

they were broken cause a bunch of stuff was missing

**Comment by hry-gh** [comment]
based change, speedmerge

**Comment by TerryTehCoder** [comment]
Why was a map file deleted for this PR? That's probably what's causing issues on your map check.

image

**Comment by hry-gh** [comment]
> Why was a map file deleted for this PR? That's probably what's causing issues on your map check.
> 
> image

it was (presumably) mistakenly added in the previous commit, it's not in the overall diff. the map checker fail is unrelated to this pr

**Review by Tsurupeta — COMMENTED** [review]
ughh, I dont want to review this...

File: code/modules/modular_computers/terminal/terminal_skill_fail.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@ GLOBAL_LIST_INIT(terminal_fails, init_subtypes(/datum/terminal_skill_fail))
 	message = "Entered id successfully banned!"
 
 /datum/terminal_skill_fail/random_ban/can_run(mob/user, datum/terminal/terminal)
-	if(!has_access(list(access_network), user.GetAccess()))
+	if(!has_access(list("ACCESS_ENGINEERING_LEVEL3"), user.GetAccess()))
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/terminal_skill_fail.dm]:
```suggestion
	if(!has_access(list(access_engineeringlvl3), user.GetAccess()))
```

File: code/modules/modular_computers/terminal/commands/status.dm [hunk 2]
```diff
@@ -7,8 +7,8 @@
 		"NOTICE: Requires network operator or admin access."
 	)
 	pattern = "^status$"
-	req_access = list(list(access_network, access_network_admin))
-	skill_needed = SKILL_EXPERT
+	req_access = list("ACCESS_ENGINEERING_LEVEL3")
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/status.dm]:
```suggestion
	req_access = list(access_engineeringlvl3)
```

File: code/modules/modular_computers/terminal/commands/ssh.dm [hunk 3]
```diff
@@ -7,13 +7,8 @@
...
 	)
 	pattern = "^ssh"
-	req_access = list(access_network_admin)
-	skill_needed = SKILL_EXPERT
-
-/datum/terminal_command/check_access(mob/user, datum/terminal/terminal)
-	if(terminal.computer.emagged() && !istype(terminal, /datum/terminal/remote))
-		return TRUE // Helps let antags do hacker gimmicks without having to get universal network access first.
-	return ..()
+	req_access = list("ACCESS_ENGINEERING_LEVEL3")
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/ssh.dm]:
```suggestion
	req_access = list(access_engineeringlvl3)
```

File: code/modules/modular_computers/terminal/commands/relays.dm [hunk 4]
```diff
@@ -7,8 +7,8 @@
 		"NOTICE: Requires network operator or admin access."
 	)
 	pattern = "^relays$"
-	req_access = list(list(access_network, access_network_admin))
-	skill_needed = SKILL_EXPERT
+	req_access = list("ACCESS_ENGINEERING_LEVEL3")
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/relays.dm]:
```suggestion
	req_access = list(access_engineeringlvl3)
```

File: code/modules/modular_computers/terminal/commands/purge.dm [hunk 5]
```diff
@@ -7,8 +7,8 @@
 		"NOTICE: Requires network admin access."
 	)
 	pattern = "^purge$"
-	req_access = list(access_network_admin)
-	skill_needed = SKILL_PROF
+	req_access = list("ACCESS_ENGINEERING_LEVEL3")
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/purge.dm]:
```suggestion
	req_access = list(access_engineeringlvl3)
```

File: code/modules/modular_computers/terminal/commands/log.dm [hunk 6]
```diff
@@ -25,6 +25,6 @@
 	M.source = S.login
 	if(!S.send_mail(argument, M))
 		return "[name]: Error; could not send email to '[argument]'."
-	if(!has_access(list(access_network_admin), user.GetAccess()))
+	if(!has_access(list("ACCESS_ENGINEERING_LEVEL3"), user.GetAccess()))
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/log.dm]:
```suggestion
	if(!has_access(list(access_engineeringlvl3), user.GetAccess()))
```

File: code/modules/modular_computers/terminal/commands/log.dm [hunk 7]
```diff
@@ -7,8 +7,8 @@
 		"NOTICE: Requires network operator or admin access. Use by non-admins is logged."
 	)
 	pattern = "^log"
-	req_access = list(list(access_network, access_network_admin))
-	skill_needed = SKILL_EXPERT
+	req_access = list("ACCESS_ENGINEERING_LEVEL3")
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/log.dm]:
```suggestion
	req_access = list(access_engineeringlvl3)
```

File: code/modules/modular_computers/terminal/commands/locate.dm [hunk 8]
```diff
@@ -19,7 +19,7 @@
 	var/nid = text2num(arguments[1])
 	if(!nid)
 		return "[name]: Error; invalid network id."
-	if(!has_access(list(access_network_admin), user.GetAccess()))
+	if(!has_access(list("ACCESS_ENGINEERING_LEVEL3"), user.GetAccess()))
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/locate.dm]:
```suggestion
	if(!has_access(list(access_engineeringlvl3), user.GetAccess()))
```

File: code/modules/modular_computers/terminal/commands/locate.dm [hunk 9]
```diff
@@ -7,8 +7,8 @@
 		"NOTICE: Requires network operator or admin access. Use by non-admins is logged."
 	)
 	pattern = "locate"
-	req_access = list(list(access_network, access_network_admin))
-	skill_needed = SKILL_PROF
+	req_access = list("ACCESS_ENGINEERING_LEVEL3")
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/locate.dm]:
```suggestion
	req_access = list(access_engineeringlvl3)
```

File: code/modules/modular_computers/terminal/commands/ban.dm [hunk 10]
```diff
@@ -9,34 +9,33 @@
...
-				return "[name]: Error; network id invalid or already banned."
-		else if(arguments[1] == "-u")
-			if (nid in ntnet_global.banned_nids)
-				ntnet_global.banned_nids -= nid
-				return "[name]: Network id '[nid]' unbanned."
-			else
-				return "[name]: Error; network id not found on list of banned ids."
-	return syntax_error()
+
+	if(!has_access(list("ACCESS_ENGINEERING_LEVEL3"), user.GetAccess()))
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/ban.dm]:
```suggestion
	if(!has_access(list(access_engineeringlvl3), user.GetAccess()))
```

File: code/modules/modular_computers/terminal/commands/ban.dm [hunk 11]
```diff
@@ -9,34 +9,33 @@
 		"NOTICE: Requires network operator access for viewing, and admin access for modification"
 	)
 	pattern = "^banned"
-	req_access = list(list(access_network, access_network_admin))
-	skill_needed = SKILL_ADEPT
+	req_access = list("ACCESS_ENGINEERING_LEVEL3")
```
Comment (Tsurupeta) [inline code/modules/modular_computers/terminal/commands/ban.dm]:
```suggestion
	req_access = list(access_engineeringlvl3)
```

**Review by Tsurupeta — CHANGES_REQUESTED** [review]
Don't use strings ffs

File: code/modules/modular_computers/terminal/terminal_commands.dm [hunk 12]
```diff
@@ -40,6 +43,24 @@ GLOBAL_LIST_INIT(terminal_commands, init_subtypes(/datum/terminal_command))
...
 	))
 	return list("Command not understood.", message)
+
+/// Returns list of arguments (if any), or null on syntax error
+/datum/terminal_command/proc/get_arguments(text)
+	var/list/arguments = splittext(text, " ")
+	if(!arguments.len || arguments[1] != name)
+		return
+
+	if(arguments.len == 1)
```
Comment (ivanmixo) [inline code/modules/modular_computers/terminal/terminal_commands.dm]:
```suggestion
	if(!length(arguments) || arguments[1] != name)
		return

	if(length(arguments) == 1)
```

File: code/modules/modular_computers/ntos/ntos.dm [hunk 13]
```diff
@@ -79,44 +79,68 @@
...
+		return
+	if(P.requires_ntnet && !get_ntnet_status(P.requires_ntnet_feature))
+		loud && show_error(user, "Unable to establish a working network connection. Please try again later. If problem persists, please contact your system administrator.")
 		return
 
-	if(P.requires_ntnet && !get_ntnet_status())
-		to_chat(user, SPAN_WARNING("Unable to establish a working network connection. Please try again later. If problem persists, please contact your system administrator."))
+	if(P in running_programs)
+		return P
+	if(running_programs.len >= get_program_capacity())
```
Comment (ivanmixo) [inline code/modules/modular_computers/ntos/ntos.dm]:
```suggestion
	if(length(running_programs) >= get_program_capacity())
```

**Review by ivanmixo — CHANGES_REQUESTED** [review]
Nitpick but `length()` is just better.

**Comment by checkraisefold** [comment]
the commit history is sublime

**Review by ivanmixo — APPROVED** [review]

**Comment by ivanmixo** [comment]
Wait til you found out batch committing suggestions exists.

**Comment by checkraisefold** [comment]
> Wait til you found out batch committing suggestions exists.

notice the 4 i did manually then the "update access" one? yeah thats exactly what happened

**Comment by ivanmixo** [comment]
Good job bestie

**Comment by Tsurupeta** [comment]
Hello, merge conflicts department?

**Comment by checkraisefold** [comment]
> Hello, merge conflicts department?

SOMEONE removed everything to do with the terminal, this will remain open until it gets fixed

File: code/modules/modular_computers/terminal/commands/file.dm [hunk 14]
```diff
@@ -4,7 +4,6 @@
 	man_entry = list(
 		"Format: file \[-flag filename \[filename2\]\]",
 		"Without options, list all files on storage devices.",
-		"With -h followed by filename, toggle the hidden flag on the file.",
```
Comment (TheTitanicEngineer) [inline code/modules/modular_computers/terminal/commands/file.dm:7]: Just wondering why was this one removed?

Comment (checkraisefold) [reply code/modules/modular_computers/terminal/commands/file.dm:7]: hidden files do not exist in our scuffed half broken modular computer code and i cba to port them

**Pull request closed without merging**